### PR TITLE
Create Python script to export to GEE

### PR DESCRIPTION
**Pull request by jterry64**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [X] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [X] Check the commit's or even all commits' message styles matches our requested structure.
- [X] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

Issue Number: GTC-2851

## What is the new behavior?

Add option to export COGs to GEE by created copy to GCS and creating a COG-backed asset using the GEE API. 

COGs are stored in the forma-250 GCS project in the bucket `data-api-gee-assets`. They have the naming convention `{dataset}/{implementation}.tif`. The GEE assets are in the forma-250 GEE project with the asset ID `{dataset}/{implementation}`.

I didn't implement versioning to start with, since that might quickly use up a lot of space in GCS/GEE and would be confusing to GEE users. So for now, assuming the asset is simply using the latest version in the API. I'm thinking we can revisit this in the future if there's a use case for better versioning.

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: batch/python/export_to_gee.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        )
+
+
+def export_to_gee(
+    dataset: str = Option(..., help="Dataset name."),
+    implementation: str = Option(..., help="Implementation name."),
+):
+    service_account = set_google_application_credentials()
+    gcs_path = upload_cog_to_gcs(dataset, implementation)
+    create_cog_backed_asset(gcs_path, dataset, implementation, service_account)
```
Comment (danscales) [inline batch/python/export_to_gee.py:65]: This is calling create_cog_backed_asset() with arguments in the wrong order, I think?  [gcs_path should be 3rd argument, and you're not using keywords]

File: batch/scripts/get_arguments.sh [hunk 2]
```diff
@@ -100,6 +100,11 @@ do
       shift # past argument
       shift # past value
       ;;
+      -G|--export_to_gee)
+      EXPORT_TO_GEE="$2"
+      shift # past argument
+      shift # past value
+      ;;
```
Comment (dmannarino) [inline batch/scripts/get_arguments.sh:107]: I think you just want 1 shift, and store something like "TRUE" (though I don't think the actual value matters). As is I think you'll be shifting past the next argument.

Comment (dmannarino) [reply batch/scripts/get_arguments.sh:107]: > I think you just want 1 shift, and store something like "TRUE" (though I don't think the actual value matters). As is I think you'll be shifting past the next argument.

If it's confusing, see the --overwrite example below.

File: batch/python/export_to_gee.py (hunk 1 above)
Comment (jterry64) [reply batch/python/export_to_gee.py]: fixed

File: batch/scripts/get_arguments.sh (hunk 2 above)
Comment (jterry64) [reply batch/scripts/get_arguments.sh:106]: oh good catch, fixed

**Review by danscales — APPROVED** [review]

**Review by dmannarino — APPROVED** [review]

**Review by solomon-negusse — APPROVED** [review]

**Pull request merged**